src/client/Microsoft.Identity.Client.Broker/WamAdapters.cs

```diff
@@ -322,7 +322,9 @@ private static string GetExpectedRedirectUri(string clientId)
         {
             //MSAL Runtime throws an ApiContractViolation Exception with Tag: 0x2039c1cb (InvalidArg)
             //When no scopes are passed, this will check if user is passing scopes
-            if (!authenticationRequestParameters.HasScopes)
+            var scopes = authenticationRequestParameters.Scope.ElementAtOrDefault(0);
+
+            if (string.IsNullOrWhiteSpace(scopes))
```

Comment:
nit, use IsNullOrEmpty
There is a slight performance improvement and whitespaces will not return true when using this

Reply:
we do want to throw for this as well, s_scopes = new[] { "   " }

Reply:
ok, makes sense